### PR TITLE
Fix creation of directories with semicolons in the name

### PR DESCRIPTION
**Pull request by sonapraneeth-a**
Creating a directory having a colon in its name (Example: 'neural-networks-deep-learning/notebook/Week 4/Deep Neural Network Application: Image Classification') (obtained from python notebook URL) is not possible in Windows. Hence we replace such a character
in the folder name with a '-' so that directory can be created in Windows

🚨Please review the [guidelines for contributing](CONTRIBUTING.md) to this repository.

## Proposed changes

Tries to fix the bug raised in [#648](https://github.com/coursera-dl/coursera-dl/issues/648)

Whenever there is a colon in the name of the folder to be created, colon is replaced by a '-' using which the folder can be created

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask. We're here
to help! This is simply a reminder of what we are going to look for before
merging your code._

- [x] I have read the [CONTRIBUTING](/CONTRIBUTING.md) doc
- [x] I agree to contribute my changes under the project's [LICENSE](/LICENSE)
- [x] I have checked that the unit tests pass locally with my changes
- [x] I have checked the style of the new code (lint/pep).
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, please explain why you chose
the solution you did and what alternatives you considered, etc.

### Reviewers
If you know the person who can review your code please add a @mention.
@rbrito

### CODE REVIEW

**Comment by rbrito** [comment]
@sonapraneeth-a, thanks a lot for discovering this problem and proposing a fix.

Can you use the function `clean_filename` that I created for this purpose? It is on the file `coursera/utils.py`.

After you do that, I will readily merge your pull request.

**Comment by sonapraneeth-a** [comment]
@rbrito I think this issue would not just be specific to Windows. It could be on other OS too. So, the title can be changed to **Fix creation of directories with semicolons**

**Comment by sonapraneeth-a** [comment]
I'll update the code soon and get back to you in case of queries.

**Comment by sonapraneeth-a** [comment]
Made changes as per your suggestion @rbrito . It looks like some commands are failing for the tests. Could you please look into them?

**Comment by balta2ar** [comment]
The build on Travis is fine, tests have passed: https://travis-ci.org/coursera-dl/coursera-dl/jobs/390060998
But it's coverage that went down, that why it's red.
As for appveyor, it's just broken for some reason, need to investigate...

File: coursera/api.py [hunk 1]
```diff
@@ -493,8 +493,14 @@ def _get_notebook_folder(self, url, jupyterId, **kwargs):
 
                 if os.path.isdir(self._course_name + "/notebook/" + head + "/") == False:
                     logging.info('Creating [{}] directories...'.format(head))
-                    os.makedirs(self._course_name + "/notebook/" + head + "/")
 
+                # head = head.replace(':', '-')
```
Comment (balta2ar) [inline coursera/api.py]: Why leave dead code?

File: coursera/api.py [hunk 2]
```diff
@@ -493,8 +493,14 @@ def _get_notebook_folder(self, url, jupyterId, **kwargs):
...
+                # To prevent problems in folder/file creation when names contain restricted symbols
+                head = clean_filename(head)
+                logging.ingo("Head: {}", head)
```
Comment (balta2ar) [inline coursera/api.py]: 1. inGo. Typo.
2.  Why do you need to print this to the user at all?
3. Did you look around the code? We're using "%s" in logging formatting. Be consistent with the current codebase.

**Comment by balta2ar** [comment]
I'm closing this pull request in favor of this one: https://github.com/coursera-dl/coursera-dl/pull/654

**Pull request closed without merging**